pkg/libartifact/store/store.go

```diff
@@ -311,38 +311,93 @@ func (as ArtifactStore) Add(ctx context.Context, dest string, paths []string, op
 	return &artifactManifestDigest, nil
 }
 
-// Inspect an artifact in a local store
-func (as ArtifactStore) Extract(ctx context.Context, nameOrDigest string, target string, options *libartTypes.ExtractOptions) error {
+func getArtifactAndImageSource(ctx context.Context, as ArtifactStore, nameOrDigest string, options *libartTypes.FilterBlobOptions) (*libartifact.Artifact, types.ImageSource, error) {
 	if len(options.Digest) > 0 && len(options.Title) > 0 {
-		return errors.New("cannot specify both digest and title")
+		return nil, nil, errors.New("cannot specify both digest and title")
 	}
 	if len(nameOrDigest) == 0 {
-		return ErrEmptyArtifactName
+		return nil, nil, ErrEmptyArtifactName
 	}
 
 	artifacts, err := as.getArtifacts(ctx, nil)
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
 
 	arty, nameIsDigest, err := artifacts.GetByNameOrDigest(nameOrDigest)
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
 	name := nameOrDigest
 	if nameIsDigest {
 		name = arty.Name
 	}
 
 	if len(arty.Manifest.Layers) == 0 {
-		return fmt.Errorf("the artifact has no blobs, nothing to extract")
+		return nil, nil, fmt.Errorf("the artifact has no blobs, nothing to extract")
 	}
 
 	ir, err := layout.NewReference(as.storePath, name)
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
 	imgSrc, err := ir.NewImageSource(ctx, as.SystemContext)
+	return arty, imgSrc, err
+}
+
+// BlobMountPaths allows the caller to access the file names from the store and how they should be mounted.
+func (as ArtifactStore) BlobMountPaths(ctx context.Context, nameOrDigest string, options *libartTypes.BlobMountPathOptions) ([]libartTypes.BlobMountPath, error) {
+	arty, imgSrc, err := getArtifactAndImageSource(ctx, as, nameOrDigest, &options.FilterBlobOptions)
+	if err != nil {
+		return nil, err
+	}
+	defer imgSrc.Close()
+
+	if len(options.Digest) > 0 || len(options.Title) > 0 {
+		digest, err := findDigest(arty, &options.FilterBlobOptions)
+		if err != nil {
+			return nil, err
+		}
+		// In case the digest is set we always use it as target name
+		// so we do not have to get the actual title annotation form the blob.
+		// Passing options.Title is enough because we know it is empty when digest
+		// is set as we only allow either one.
+		filename, err := generateArtifactBlobName(options.Title, digest)
+		if err != nil {
+			return nil, err
+		}
+		path, err := layout.GetLocalBlobPath(ctx, imgSrc, digest)
```

Comment:
I completely forgot to think about this previously: the path will be absolute iff the input path to `layout.NewReference` is absolute. I hope that’s fine.

Reply:
Dealing with relative paths seems unsafe I am not sure if the oci runtimes even accepts relative paths as source of a bind mount. Even if they are it is not guaranteed that we never change the cwd.

I guess I can add a sanity check here with filepath.Abs() an error if the path is no absolute as it should not happen.

Reply:
Per this PR, the path comes from `runtime.storageConfig.GraphRoot` (and I didn’t go trace that further). I _guess_ that if that were relative, a lot would break.

But, also, it’s so much easier to stick a `filepath.Abs` into `NewArtifactStore ` constructor than to trace the origins and consequences, that I think it’s worth the peace of mind.

Reply:
Added a commit to validate the path in NewArtifactStore().